project/src/main/java/com/google/sps/data/Places.java

```diff
@@ -38,5 +45,40 @@
     return ImmutableList.copyOf(mutablePlaces);
   }
 
+  /**
+   * filters the given list of places according to the given parameters.
+   * @param places the list we want to filter.
+   * @param minRating The minimal rating we want to limit the list by. A number between 1 and 5.
+   * @param filterIfNoWebsite Specifies whether we should filter if there is no available website
+   *        for the restaurant.
+   * @param filterBranchesOfSamePlace Specifies whether we should remove different branches of the
+   *        same place from the results. Places are considered the same if they have the same name.
+   * @return the filtered list.
+   */
+  public static ImmutableList<Place> filter(ImmutableList<Place> places, int minRating, Boolean
+      filterIfNoWebsite, Boolean filterBranchesOfSamePlace) {
+    //TODO(M1): also filter by place's status when the attribute is added
+    ImmutableList<Place> result = ImmutableList.copyOf(
+        places.stream()
+            .sorted(Comparator.comparing(Place::rating).reversed())
+            .filter(place -> place.rating() >= minRating)
+      //TODO(M1): take in account whether the place has a google maps link, when attribute is added
+            .filter(filterIfNoWebsite
+                ? place -> !Strings.isNullOrEmpty(place.websiteUrl())
+                : place -> true)
+            .filter(filterBranchesOfSamePlace
```

Comment:
This one can be written as filter(filterBranchesOfSamePlace && distinctByKey(Place::name))
same for the other filter.

Reply:
On a different note, I'm not a fan of the distinctByKey usage. It's a hard read and I had to lookup an explanation myself to figure out what's going on here. For readability's sake, I propose you do the distinct phase using a set of some sort (similar to here: https://stackoverflow.com/questions/29670116/remove-duplicates-from-a-list-of-objects-based-on-property-in-java-8) or just separate the distinct operation from the one-liner.

Reply:
The second comment - done. 
The first comment - I'm not sure it makes sense to me. The function in the filter returns true when we want to keep the item, and false when we don't want to keep it. If I use this condition, then when we don't want to filter by the website (AKA when filterIfNoWebsite is false), the && condition will always be false and we'll filter everything out.

Reply:
My bad, forgot to add an "!",see this condition: !(filterIfNoWebsite && Strings.isNullOrEmpty(place.websiteUrl()))
You can write down all cases and see it answers all scenarios correctly.

You can logically think about this as, what are the cases here (4 options for 2 parameters), then list down the desired outcome on each case. You'll then notice that the filter function should return false on only one case (filterIfNoWebsite == True and Strings.isNullOrEmpty(place.websiteUrl()) == True). The remaining cases should not filter this out.

Hope that helps:) Notice that what triggered my comment is that if you listed the filter condition as an if-else clause where one of the clauses just returns a boolean value - it can be simplified to one expression for sure.

Reply:
ok got it, thanks! done